TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/KafkaPrinter.java

```diff
@@ -230,31 +253,47 @@ static void processNextChunkOfKafkaEvents(Consumer<String, byte[]> kafkaConsumer
         binaryReceiver.accept(StreamSupport.stream(records.spliterator(), false));
     }
 
-    static java.util.function.Consumer<Stream<ConsumerRecord<String, byte[]>>> getDelimitedProtoBufOutputter(OutputStream outputStream, Map<Partition, PartitionTracker> capturedRecords) {
-        CodedOutputStream codedOutputStream = CodedOutputStream.newInstance(outputStream);
+    static java.util.function.Consumer<Stream<ConsumerRecord<String, byte[]>>> getDelimitedProtoBufOutputter(Map<TopicPartition, PartitionTracker> capturedRecords,
+        Map<Integer, CodedOutputStream> partitionOutputStreams, boolean separatePartitionOutputs) {
```

Comment:
I should have asked before, but why are you using a CodedOutputStream (full disclosure, I wrote it with a CodedOutputStream) here instead of [writeDelimitedTo](https://protobuf.dev/reference/java/api-docs/com/google/protobuf/MessageLite.html#writeDelimitedTo-java.io.OutputStream-)?  That would let you pass in a compressed output stream and you wouldn't need to worry about the uncompressed buffering problem.

Reply:
I like the idea of using this `writeDelimitedTo` function but it seems to be on the Message interface, which for us it seems we would have to take the byte[] and have our TrafficStream protobuf object parse it, then we could use `writeDelimitedTo`. I wasn't sure after these requirements if there would be much benefit

Reply:
Ahhh - good point.  The current code is probably more efficient, so until we have a reason to, there's probably not a good reason to make the change now.